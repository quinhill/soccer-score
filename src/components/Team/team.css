.team-container {
  display: grid;
  height: 100%;
  width: 100%;
  grid-template-columns: 50% 50%;
  grid-template-rows: 20% 40% 40%
}

.team {
  grid-column: 1 / 3;
  grid-row: 1 / 2;
  margin: 30px;
  color: white;
  display: flex;
  width: 100%;
  display: flex;
  flex-direction: column;
  align-items: center;
}

#team-logo {
  height: 150px;
  width: 150px;
  margin-bottom: 10px;
}

h1 {
  margin: 0 5px 0 0;
  font-size: 20px;
}

h3 {
  margin: 0 5px;
  color: white;
}

.manager-stadium {
  grid-column: 1 / 2;
  grid-row: 2 / 3;
  background-color: #b5b5b7;
  margin-right: 10px;
  border-radius: 5px;
  box-shadow: 2px 2px 10px 0px rgba(0, 0, 0, 0.6);
}

.manager-container {
  display: flex;
  justify-content: center;
  align-items: flex-end;
  margin-top: 10px;
}

span {
  margin-right: 10px;
}

.manager {
  color: #252839;
  margin: 0 10px;
}

.stadium-container {
  display: grid;
  grid-template-rows: 50% 50%;
}

.stadium {
  color: #252839;
}

.players {
  grid-column: 2 / 3;
  grid-row: 2 / 4;
  width: 400px;
  background-color: #b5b5b7;
  border-radius: 5px;
  padding-top: 10px;
  margin-bottom: 10px;
  box-shadow: 2px 2px 10px 0px rgba(0, 0, 0, 0.6);
}

.player {
  display: grid;
  padding-top: 10px;
  height: 20px;
  grid-template-columns: 20% 80%;
  text-align: start;
  border-bottom: 1px solid white;
}

.player:last-child {
  border: none;
}

.player-link {
  color: #252839;
  text-decoration: none;
}

.player-link:hover {
  color: #ffffff;
}

.player-number {
  color: #252839;
  margin: 0 0 0 20px;
}