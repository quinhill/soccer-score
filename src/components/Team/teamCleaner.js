export const cleanTeam = (squad) => {
  const players = squad.players.map(player => (
    {
      name: player.name
    }
  ))
  return {
    teamName: squad.team.name,
    manager: {
      name: squad.manager.name,
      nationality: squad.manager.nationality
    },
    players,
    country: squad.team.country,
    stadium: {
      name: squad.venue.name,
      capacity: squad.venue.capacity
    }
  }
}

const teamObj = {
  manager: {
    country_code: "ARG",
    id: "sr:player:1018137",
    name: "Grioni, Marcelo",
    nationality: "Argentina"
  },
  players: [
    {
      country_code:"PER",
      date_of_birth:"1999-06-30",
      gender:"male",
      height:180,
      id:"sr:player:1408961",
      name:"Fuentes Pando, Gibson",
      nationality:"Peru",
      preferred_foot:"right",
      type:"forward",
      weight:76
    },
    { id: "sr:player:1017993", name: "Lliuya, Marcos", type: "midfielder", date_of_birth: "1992-03-27", nationality: "Peru"},
    { id: "sr:player:1408621", name: "Saune, Jonathan", type: "forward", date_of_birth: "1989-07-20", nationality: "Peru"},
    { id: "sr:player:1478884", name: "Menacho, Rodrigo", type: "defender", date_of_birth: "1999-04-13", nationality: "Peru"},
    { id: "sr:player:556856", name: "Salcedo, Ricardo", type: "midfielder", date_of_birth: "1990-03-23", nationality: "Peru"},
    { id: "sr:player:41714", name: "Pena, Victor", type: "midfielder", date_of_birth: "1987-05-14", nationality: "Peru"},
    { id: "sr:player:248243", name: "Monsalvo, Charles", type: "forward", date_of_birth: "1990-05-26", nationality: "Colombia"},
    { id: "sr:player:189207", name: "Deza, Jean", type: "forward", date_of_birth: "1993-06-09", nationality: "Peru"},
    { id: "sr:player:47413", name: "Trujillo, Luis", type: "midfielder", date_of_birth: "1990-12-27", nationality: "Peru"},
    { id: "sr:player:1425985", name: "Velasquez, Moises", type: "defender", date_of_birth: "1999-02-24", nationality: "Peru"},
    { id: "sr:player:937128", name: "Lojas, Sebastian", type: "defender", date_of_birth: "1995-07-04", nationality: "Peru"},
    { id: "sr:player:162425", name: "Hermoza, Eder", type: "goalkeeper", date_of_birth: "1990-04-04", nationality: "Peru"},
    { id: "sr:player:73666", name: "Valverde, Marcio Andres", type: "midfielder", date_of_birth: "1987-10-23", nationality: "Peru"},
    { id: "sr:player:1108885", name: "Calderon, Waldir", type: "midfielder", date_of_birth: "1997-06-09", nationality: "Peru"},
    { id: "sr:player:125040", name: "Pinto, Joel", type: "goalkeeper", date_of_birth: "1980-06-05", nationality: "Peru"}
  ],
  team: {
    abbreviation:"HUA",
    category:{ id: "sr:category:20", name: "Peru", country_code: "PER" },
    country:"Peru",
    country_code:"PER",
    id:"sr:competitor:33895",
    name: "Sport Huancayo"
  },
  venue: {
    capacity: 20000,
    city_name: "Huancayo",
    country_code: "PER",
    country_name: "Peru",
    id: "sr:venue:1175",
    map_coordinates: "-12.072796,-75.201588",
    name: "Estadio Huancayo"
  }
}