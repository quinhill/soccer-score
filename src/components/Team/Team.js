import React from 'react';
import { connect } from 'react-redux';
import './team.css';
import PropTypes from 'prop-types';

export const Team = (props) => {

  let players;
  if (props.players) {
    players = props.players.map((player, index) => {
      return (
        <div className="player">
          <p key={index}>{player.name}</p>
          <p>{player.position}</p>
        </div>
      )
    })
  }
  
  return (
    <div className="team-container">
      <div className="team">
        <h1>{props.name}</h1>
        <h3>{props.country}</h3>
      </div>
      <div className="manager-stadium">
        <div className="manager-container">
          <p className="manager">manager:</p>
          <h3>{props.manager},</h3>
          <h3>{props.managerNationality}</h3>
        </div>
        <div className="stadium-container">
          <p className="stadium">stadium:</p>
          <p>{props.stadiumName}</p>
          <p>capacity:</p>
          <p>{props.stadiumCapacity}</p>
        </div>
      </div>
      <div className="players">
        {players}
      </div>
    </div>
  )
}

Team.propTypes = {
  squad: PropTypes.object
}

export const mapStateToProps = (state) => ({
  name: state.squad.teamName,
  manager: state.squad.managerName,
  managerNationality: state.squad.managerNationality,
  players: state.squad.players,
  country: state.squad.country,
  stadiumName: state.squad.stadiumName,
  stadiumCapacity: state.squad.stadiumCapacity
})

export default connect(mapStateToProps)(Team);