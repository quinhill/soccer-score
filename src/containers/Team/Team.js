import React, { Component } from 'react';
import './team.css';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import { getPlayerUrl } from '../../apiKeys';
import { fetchPlayer } from '../../thunks/fetchPlayer'
import { Link } from 'react-router-dom';
import { setDisplay } from '../../actions/setDisplayAction';

export class Team extends Component {
  constructor(props) {
    super(props)
    this.state = {
      player: false
    }
  }

  getPlayer = (id) => {
    const url = getPlayerUrl(id)
    this.props.fetchPlayer(url)
    this.props.setDisplay('player')
  }

  mappedPlayers = () => {
    return this.props.squad.map(player => {
      return (
        <div className="player">
          <p className="player-number">{player.number}</p>
          <Link
            to='/player'
            className="player-link"
            onClick={() => this.getPlayer(player.id)}
          >
            {player.name}
          </Link>
        </div>
      )
    })
  }

  render () {
    return (
      <div className="container-container">
        <div className="team-container">
          <div className="team">
            <img id="team-logo" src={this.props.team.logo} />
            <h1 className="team-name">{this.props.team.name}</h1>
          </div>
          <div className="manager-stadium">
            <div className="manager-container">
              <h3 className="manager"><span>Manager:</span>{this.props.team.manager},</h3>
              <h3 className="manager">{this.props.team.nationality}</h3>
            </div>
            <div className="stadium-container">
              <p className="stadium"><span>Stadium:</span>{this.props.team.venue}</p>
              <p className="stadium"><span>Capacity:</span>{this.props.team.capacity}</p>
            </div>
          </div>
          <div className="players">
            <h3></h3>
            {this.mappedPlayers()}
          </div>
        </div>
      </div>
    )
  }
}

Team.propTypes = {
  team: PropTypes.object,
  squad: PropTypes.array,
  fetchPlayer: PropTypes.func,
  setDisplay: PropTypes.func
}

export const mapStateToProps = state => ({
  team: state.team,
  squad: state.squad
})

export const mapDispatchToProps = dispatch => ({
  fetchPlayer: (url) => dispatch(fetchPlayer(url)),
  setDisplay: (clicked) => dispatch(setDisplay(clicked))
})

export default connect(mapStateToProps, mapDispatchToProps)(Team)