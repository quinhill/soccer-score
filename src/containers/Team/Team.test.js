import React from 'react';
import { Team } from './Team';
import { shallow } from 'enzyme';
import { mapStateToProps, mapDispatchToProps } from './Team';

describe('Team', () => {
  it('should match snapshot', () => {
    const mockSquad = [{}, {}];
    const mockTeam = {};
    const mockFetchPlayer = jest.fn();
    const mockSetDisplay = jest.fn();
    const wrapper = shallow(<Team
      squad={mockSquad}
      team={mockTeam}
      fetchPlayer={mockFetchPlayer}
      setDisplay={mockSetDisplay}
      />);
    
    expect(wrapper).toMatchSnapshot();
  })

  describe('mapStateToProps', () => {
    it('should return a props object', () => {
      const initialState = {
        player: {id: 'some player'},
        squad: [{}, {}],
        team: {id: 'some string'}
      }
      const expected = {
        squad: [{}, {}],
        team: {id: 'some string'}
      }
  
      const result = mapStateToProps(initialState);
  
      expect(result).toEqual(expected);
    })
  })

  describe('mapDispatchToProps', () => {
    it('should dispatch a fetchPlayer action when fetchPlayer is called', () => {
      const mockDispatch = jest.fn()
      const url = 'http://somestring.com';

      const mappedProps = mapDispatchToProps(mockDispatch)
      mappedProps.fetchPlayer(url)

      expect(mockDispatch).toHaveBeenCalled()
    })
    it('should dispatch a setDisplay action when setDisplay is called', () => {
      const mockDispatch = jest.fn()
      const clicked = 'player';

      const mappedProps = mapDispatchToProps(mockDispatch)
      mappedProps.setDisplay(clicked)

      expect(mockDispatch).toHaveBeenCalled()
    })
  })
})