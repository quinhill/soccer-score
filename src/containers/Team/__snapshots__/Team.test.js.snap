// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Team should match snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Team
    fetchPlayer={[Function]}
    setDisplay={[Function]}
    squad={
        Array [
            Object {},
            Object {},
          ]
    }
    team={Object {}}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <div
        className="team-container"
>
        <div
                className="team"
        >
                <img
                        id="team-logo"
                        src={undefined}
                />
                <h1
                        className="team-name"
                />
        </div>
        <div
                className="manager-stadium"
        >
                <div
                        className="manager-container"
                >
                        <h3
                                className="manager"
                        >
                                <span>
                                        Manager:
                                </span>
                                ,
                        </h3>
                        <h3
                                className="manager"
                        />
                </div>
                <div
                        className="stadium-container"
                >
                        <p
                                className="stadium"
                        >
                                <span>
                                        Stadium:
                                </span>
                        </p>
                        <p
                                className="stadium"
                        >
                                <span>
                                        Capacity:
                                </span>
                        </p>
                </div>
        </div>
        <div
                className="players"
        >
                <h3 />
                <div
                        className="player"
                >
                        <p
                                className="player-number"
                        />
                        <Link
                                className="player-link"
                                onClick={[Function]}
                                replace={false}
                                to="/player"
                        />
                </div>
                <div
                        className="player"
                >
                        <p
                                className="player-number"
                        />
                        <Link
                                className="player-link"
                                onClick={[Function]}
                                replace={false}
                                to="/player"
                        />
                </div>
        </div>
</div>,
      "className": "container-container",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="team"
>
            <img
                        id="team-logo"
                        src={undefined}
            />
            <h1
                        className="team-name"
            />
</div>,
          <div
            className="manager-stadium"
>
            <div
                        className="manager-container"
            >
                        <h3
                                    className="manager"
                        >
                                    <span>
                                                Manager:
                                    </span>
                                    ,
                        </h3>
                        <h3
                                    className="manager"
                        />
            </div>
            <div
                        className="stadium-container"
            >
                        <p
                                    className="stadium"
                        >
                                    <span>
                                                Stadium:
                                    </span>
                        </p>
                        <p
                                    className="stadium"
                        >
                                    <span>
                                                Capacity:
                                    </span>
                        </p>
            </div>
</div>,
          <div
            className="players"
>
            <h3 />
            <div
                        className="player"
            >
                        <p
                                    className="player-number"
                        />
                        <Link
                                    className="player-link"
                                    onClick={[Function]}
                                    replace={false}
                                    to="/player"
                        />
            </div>
            <div
                        className="player"
            >
                        <p
                                    className="player-number"
                        />
                        <Link
                                    className="player-link"
                                    onClick={[Function]}
                                    replace={false}
                                    to="/player"
                        />
            </div>
</div>,
        ],
        "className": "team-container",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <img
                id="team-logo"
                src={undefined}
/>,
              <h1
                className="team-name"
/>,
            ],
            "className": "team",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "id": "team-logo",
                "src": undefined,
              },
              "ref": null,
              "rendered": null,
              "type": "img",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": undefined,
                "className": "team-name",
              },
              "ref": null,
              "rendered": null,
              "type": "h1",
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <div
                className="manager-container"
>
                <h3
                                className="manager"
                >
                                <span>
                                                Manager:
                                </span>
                                ,
                </h3>
                <h3
                                className="manager"
                />
</div>,
              <div
                className="stadium-container"
>
                <p
                                className="stadium"
                >
                                <span>
                                                Stadium:
                                </span>
                </p>
                <p
                                className="stadium"
                >
                                <span>
                                                Capacity:
                                </span>
                </p>
</div>,
            ],
            "className": "manager-stadium",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <h3
                    className="manager"
>
                    <span>
                                        Manager:
                    </span>
                    ,
</h3>,
                  <h3
                    className="manager"
/>,
                ],
                "className": "manager-container",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": Array [
                      <span>
                        Manager:
</span>,
                      undefined,
                      ",",
                    ],
                    "className": "manager",
                  },
                  "ref": null,
                  "rendered": Array [
                    Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "host",
                      "props": Object {
                        "children": "Manager:",
                      },
                      "ref": null,
                      "rendered": "Manager:",
                      "type": "span",
                    },
                    undefined,
                    ",",
                  ],
                  "type": "h3",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": undefined,
                    "className": "manager",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": "h3",
                },
              ],
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <p
                    className="stadium"
>
                    <span>
                                        Stadium:
                    </span>
</p>,
                  <p
                    className="stadium"
>
                    <span>
                                        Capacity:
                    </span>
</p>,
                ],
                "className": "stadium-container",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": Array [
                      <span>
                        Stadium:
</span>,
                      undefined,
                    ],
                    "className": "stadium",
                  },
                  "ref": null,
                  "rendered": Array [
                    Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "host",
                      "props": Object {
                        "children": "Stadium:",
                      },
                      "ref": null,
                      "rendered": "Stadium:",
                      "type": "span",
                    },
                    undefined,
                  ],
                  "type": "p",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": Array [
                      <span>
                        Capacity:
</span>,
                      undefined,
                    ],
                    "className": "stadium",
                  },
                  "ref": null,
                  "rendered": Array [
                    Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "host",
                      "props": Object {
                        "children": "Capacity:",
                      },
                      "ref": null,
                      "rendered": "Capacity:",
                      "type": "span",
                    },
                    undefined,
                  ],
                  "type": "p",
                },
              ],
              "type": "div",
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <h3 />,
              Array [
                <div
                  className="player"
>
                  <p
                                    className="player-number"
                  />
                  <Link
                                    className="player-link"
                                    onClick={[Function]}
                                    replace={false}
                                    to="/player"
                  />
</div>,
                <div
                  className="player"
>
                  <p
                                    className="player-number"
                  />
                  <Link
                                    className="player-link"
                                    onClick={[Function]}
                                    replace={false}
                                    to="/player"
                  />
</div>,
              ],
            ],
            "className": "players",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {},
              "ref": null,
              "rendered": null,
              "type": "h3",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <p
                    className="player-number"
/>,
                  <Link
                    className="player-link"
                    onClick={[Function]}
                    replace={false}
                    to="/player"
/>,
                ],
                "className": "player",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": undefined,
                    "className": "player-number",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": "p",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "children": undefined,
                    "className": "player-link",
                    "onClick": [Function],
                    "replace": false,
                    "to": "/player",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
              ],
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <p
                    className="player-number"
/>,
                  <Link
                    className="player-link"
                    onClick={[Function]}
                    replace={false}
                    to="/player"
/>,
                ],
                "className": "player",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": undefined,
                    "className": "player-number",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": "p",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "children": undefined,
                    "className": "player-link",
                    "onClick": [Function],
                    "replace": false,
                    "to": "/player",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
              ],
              "type": "div",
            },
          ],
          "type": "div",
        },
      ],
      "type": "div",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <div
          className="team-container"
>
          <div
                    className="team"
          >
                    <img
                              id="team-logo"
                              src={undefined}
                    />
                    <h1
                              className="team-name"
                    />
          </div>
          <div
                    className="manager-stadium"
          >
                    <div
                              className="manager-container"
                    >
                              <h3
                                        className="manager"
                              >
                                        <span>
                                                  Manager:
                                        </span>
                                        ,
                              </h3>
                              <h3
                                        className="manager"
                              />
                    </div>
                    <div
                              className="stadium-container"
                    >
                              <p
                                        className="stadium"
                              >
                                        <span>
                                                  Stadium:
                                        </span>
                              </p>
                              <p
                                        className="stadium"
                              >
                                        <span>
                                                  Capacity:
                                        </span>
                              </p>
                    </div>
          </div>
          <div
                    className="players"
          >
                    <h3 />
                    <div
                              className="player"
                    >
                              <p
                                        className="player-number"
                              />
                              <Link
                                        className="player-link"
                                        onClick={[Function]}
                                        replace={false}
                                        to="/player"
                              />
                    </div>
                    <div
                              className="player"
                    >
                              <p
                                        className="player-number"
                              />
                              <Link
                                        className="player-link"
                                        onClick={[Function]}
                                        replace={false}
                                        to="/player"
                              />
                    </div>
          </div>
</div>,
        "className": "container-container",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <div
              className="team"
>
              <img
                            id="team-logo"
                            src={undefined}
              />
              <h1
                            className="team-name"
              />
</div>,
            <div
              className="manager-stadium"
>
              <div
                            className="manager-container"
              >
                            <h3
                                          className="manager"
                            >
                                          <span>
                                                        Manager:
                                          </span>
                                          ,
                            </h3>
                            <h3
                                          className="manager"
                            />
              </div>
              <div
                            className="stadium-container"
              >
                            <p
                                          className="stadium"
                            >
                                          <span>
                                                        Stadium:
                                          </span>
                            </p>
                            <p
                                          className="stadium"
                            >
                                          <span>
                                                        Capacity:
                                          </span>
                            </p>
              </div>
</div>,
            <div
              className="players"
>
              <h3 />
              <div
                            className="player"
              >
                            <p
                                          className="player-number"
                            />
                            <Link
                                          className="player-link"
                                          onClick={[Function]}
                                          replace={false}
                                          to="/player"
                            />
              </div>
              <div
                            className="player"
              >
                            <p
                                          className="player-number"
                            />
                            <Link
                                          className="player-link"
                                          onClick={[Function]}
                                          replace={false}
                                          to="/player"
                            />
              </div>
</div>,
          ],
          "className": "team-container",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <img
                  id="team-logo"
                  src={undefined}
/>,
                <h1
                  className="team-name"
/>,
              ],
              "className": "team",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "id": "team-logo",
                  "src": undefined,
                },
                "ref": null,
                "rendered": null,
                "type": "img",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": undefined,
                  "className": "team-name",
                },
                "ref": null,
                "rendered": null,
                "type": "h1",
              },
            ],
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <div
                  className="manager-container"
>
                  <h3
                                    className="manager"
                  >
                                    <span>
                                                      Manager:
                                    </span>
                                    ,
                  </h3>
                  <h3
                                    className="manager"
                  />
</div>,
                <div
                  className="stadium-container"
>
                  <p
                                    className="stadium"
                  >
                                    <span>
                                                      Stadium:
                                    </span>
                  </p>
                  <p
                                    className="stadium"
                  >
                                    <span>
                                                      Capacity:
                                    </span>
                  </p>
</div>,
              ],
              "className": "manager-stadium",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    <h3
                      className="manager"
>
                      <span>
                                            Manager:
                      </span>
                      ,
</h3>,
                    <h3
                      className="manager"
/>,
                  ],
                  "className": "manager-container",
                },
                "ref": null,
                "rendered": Array [
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": Array [
                        <span>
                          Manager:
</span>,
                        undefined,
                        ",",
                      ],
                      "className": "manager",
                    },
                    "ref": null,
                    "rendered": Array [
                      Object {
                        "instance": null,
                        "key": undefined,
                        "nodeType": "host",
                        "props": Object {
                          "children": "Manager:",
                        },
                        "ref": null,
                        "rendered": "Manager:",
                        "type": "span",
                      },
                      undefined,
                      ",",
                    ],
                    "type": "h3",
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": undefined,
                      "className": "manager",
                    },
                    "ref": null,
                    "rendered": null,
                    "type": "h3",
                  },
                ],
                "type": "div",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    <p
                      className="stadium"
>
                      <span>
                                            Stadium:
                      </span>
</p>,
                    <p
                      className="stadium"
>
                      <span>
                                            Capacity:
                      </span>
</p>,
                  ],
                  "className": "stadium-container",
                },
                "ref": null,
                "rendered": Array [
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": Array [
                        <span>
                          Stadium:
</span>,
                        undefined,
                      ],
                      "className": "stadium",
                    },
                    "ref": null,
                    "rendered": Array [
                      Object {
                        "instance": null,
                        "key": undefined,
                        "nodeType": "host",
                        "props": Object {
                          "children": "Stadium:",
                        },
                        "ref": null,
                        "rendered": "Stadium:",
                        "type": "span",
                      },
                      undefined,
                    ],
                    "type": "p",
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": Array [
                        <span>
                          Capacity:
</span>,
                        undefined,
                      ],
                      "className": "stadium",
                    },
                    "ref": null,
                    "rendered": Array [
                      Object {
                        "instance": null,
                        "key": undefined,
                        "nodeType": "host",
                        "props": Object {
                          "children": "Capacity:",
                        },
                        "ref": null,
                        "rendered": "Capacity:",
                        "type": "span",
                      },
                      undefined,
                    ],
                    "type": "p",
                  },
                ],
                "type": "div",
              },
            ],
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <h3 />,
                Array [
                  <div
                    className="player"
>
                    <p
                                        className="player-number"
                    />
                    <Link
                                        className="player-link"
                                        onClick={[Function]}
                                        replace={false}
                                        to="/player"
                    />
</div>,
                  <div
                    className="player"
>
                    <p
                                        className="player-number"
                    />
                    <Link
                                        className="player-link"
                                        onClick={[Function]}
                                        replace={false}
                                        to="/player"
                    />
</div>,
                ],
              ],
              "className": "players",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {},
                "ref": null,
                "rendered": null,
                "type": "h3",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    <p
                      className="player-number"
/>,
                    <Link
                      className="player-link"
                      onClick={[Function]}
                      replace={false}
                      to="/player"
/>,
                  ],
                  "className": "player",
                },
                "ref": null,
                "rendered": Array [
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": undefined,
                      "className": "player-number",
                    },
                    "ref": null,
                    "rendered": null,
                    "type": "p",
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "class",
                    "props": Object {
                      "children": undefined,
                      "className": "player-link",
                      "onClick": [Function],
                      "replace": false,
                      "to": "/player",
                    },
                    "ref": null,
                    "rendered": null,
                    "type": [Function],
                  },
                ],
                "type": "div",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    <p
                      className="player-number"
/>,
                    <Link
                      className="player-link"
                      onClick={[Function]}
                      replace={false}
                      to="/player"
/>,
                  ],
                  "className": "player",
                },
                "ref": null,
                "rendered": Array [
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": undefined,
                      "className": "player-number",
                    },
                    "ref": null,
                    "rendered": null,
                    "type": "p",
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "class",
                    "props": Object {
                      "children": undefined,
                      "className": "player-link",
                      "onClick": [Function],
                      "replace": false,
                      "to": "/player",
                    },
                    "ref": null,
                    "rendered": null,
                    "type": [Function],
                  },
                ],
                "type": "div",
              },
            ],
            "type": "div",
          },
        ],
        "type": "div",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
